pkg/controllers/nodepool/readiness/controller.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package readiness
+
+import (
+	"context"
+
+	logger "log"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/awslabs/operatorpkg/status"
+
+	"sigs.k8s.io/karpenter/pkg/utils/nodepool"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
```

Comment:
one nit i just noticed. If you put lines in between imports, the linter won't sort them. Best to do 3 groups here

1. golang native libraries
2. non sigs/karpenter libraries
3. sigs/karpenter libraries